docs: update readme to clarify toolbox server run instructions

SaxenaAnushka102 opened the pull request:
### Description

Update [README.md](https://github.com/googleapis/genai-toolbox/blob/main/README.md#running-the-server) to clarify Toolbox server run instructions for Binary, Docker, and Source installations.

### Related issue(s)

Addresses and resolves #952

SaxenaAnushka102 commented:
Hello @kurtisvg !
I'd like to follow up on this, please. Thanks!

averikitsch commented on README.md:
```diff
@@ -151,9 +151,39 @@ go install github.com/googleapis/genai-toolbox@v0.10.0
+<details open>
+<summary>Source</summary>
+
+To run Toolbox from Local Source:
```
nit: do not capitalize local source

averikitsch commented on README.md:
````diff
@@ -151,9 +151,39 @@ go install github.com/googleapis/genai-toolbox@v0.10.0
+To run Toolbox from Local Source:
+
+```sh
+go run .
````
Maybe we can just link to how to compile the binary. Is that in our Developer.md?

SaxenaAnushka102 commented:
Hello maintainers,
I apologize for the confusion with the previous commits on this PR. I ran into some issues with Git rebasing and ended up with a messy commit history. I've since cleaned up the branch to a single, clear commit that incorporates all the requested changes. The previous comments are no longer visible, but I have addressed all the feedback. Could you please take another look when you have a moment? Thanks for your patience & understanding.

averikitsch commented on README.md:
````diff
@@ -170,21 +175,61 @@ execute `toolbox` to start the server:
+--tools-file "/app/tools.yaml"
+```
+
+>[!NOTE]>The `-v` flag mounts your local `tools.yaml` into the container, and `-p` maps the container's port `5000` to your host's port `5000`.
````
The note tag is not rendering as expected. See https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

averikitsch commented on README.md:
```diff
@@ -170,21 +175,61 @@ execute `toolbox` to start the server:
+
+<summary>Homebrew Users</summary>
 
 If you installed Toolbox using Homebrew, the `toolbox` binary is available in your system path. You can start the server with the same command:
```
Can we add the homebrew download instructions or a link?

averikitsch commented on README.md:
````diff
@@ -170,21 +175,62 @@ execute `toolbox` to start the server:
+go run .
+```
+
+> [!NOTE]
````
It doesn't look like any of the notes work when using them in the drop downs. Can we plan text instead of the quote notation and maybe just bold "Note"?

SaxenaAnushka102 replied:
Sure, I'll check and update this
